### ISSUE TITLE
How to programmatically add analyzer

### ISSUE THREAD

**Issue by czhang120** [margitaii/pydeequ#8]
Hi, 

I would like to dynamically add analyzer to the analysis run (Method2) vs the static approach in the example (Method1). Somehow, I noticed that method 2 runs much slower than method 1. Any suggestions? 

Thanks for the great work to make deequ available in python and we would like to contribute. 

**Method 1**
r1 = (AnalysisRunner(spark)
...     .onData(olsmodel1.transformed_train)
...     .addAnalyzer(analyzers.Size())
...     .addAnalyzer(analyzers.Completeness('acctos_1'))
...     .addAnalyzer(analyzers.ApproxCountDistinct('acctos_1'))
...     .addAnalyzer(analyzers.Mean('acctos_1'))
...     .run())
out = DataFrame(r1, spark)

**Method 2**
r2 = r2.addAnalyzer(analyzers.Size())
r2 = r2.addAnalyzer(analyzers.Completeness('acctos_1'))
r2 = r2.addAnalyzer(analyzers.ApproxCountDistinct('acctos_1'))
r2 = r2.addAnalyzer(analyzers.Mean('acctos_1'))
r2.run()

**Comment by margitaii** [comment]
Hi @czhang120 , sorry for the late reply I'll come back to you in the next days with some investigation regarding the difference in performance.
The main reason behind the "static approach" was to follow the API design of the original Deequ written in Scala but depending on the need of python programmers I am open to change it.

**Comment by czhang120** [comment]
Thanks! We found a fix to the issue but not sure why it happened. In our first try, we ran method 1 first and method 2 next in the same jupyter session and noted that the second method took much longer to finish. Then we re-started the jupyter session and only executed the method 2 and the completion time is about the same as running method 1. The reason we are testing method 2 is to have a for-loop to dynamically add analyzers to Analysis Runner.
